packages/vaadin-combo-box/src/vaadin-combo-box-dropdown.js

```diff
@@ -221,6 +221,13 @@ class ComboBoxDropdownElement extends DisableUpgradeMixin(mixinBehaviors(IronRes
     return this.alignedAbove ? -overlayRect.height : targetRect.height;
   }
 
+  _shouldAlignLeft(targetRect) {
+    const spaceRight =
+      (window.innerWidth - targetRect.right - Math.min(document.body.scrollLeft, 0)) / window.innerWidth;
```

Comment:
I wonder if `scrollLeft` takes negative when `direction=rtl`.

> If the element's direction is rtl (right-to-left), then scrollLeft is 0 when the scrollbar is at its rightmost position (at the start of the scrolled content), and then increasingly negative as you scroll towards the end of the content.

https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollLeft

Reply:
If this is the case, I would suggest creating another test case to cover the `rtl` direction.

Reply:
...if this is the case, it might be worthy of creating another test case to cover `direction=rtl`.

Reply:
OK thanks, that would explain how it can assume a negative value.

However I'm still sceptical of the logic, as bounding client rect should already compensate for the scroll position: https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect#value

I might just have to test this, I'm having a hard time visualising this.

Reply:
It turns out that it's not required to cover the rtl case. Removed.